lib/HeroBonus.cpp

```diff
@@ -809,8 +802,16 @@ void CBonusSystemNode::updateBonuses(const CSelector &s)
 		child->updateBonuses(s);
 }
 
-void CBonusSystemNode::addNewBonus(Bonus *b)
+void CBonusSystemNode::addNewBonus(const std::shared_ptr<Bonus>& b)

```

Comment:
Okay, I tried making a copy and vcmiserver started to segfault in Template dctor at game end. Bad idea. Damn, it is so fragile!